common/djangoapps/student/helpers.py

```diff
@@ -122,8 +124,16 @@ def check_verify_status_by_course(user, course_enrollment_pairs, all_course_mode
         # verification status.
         if verified_mode is not None and enrollment.mode in CourseMode.VERIFIED_MODES:
             deadline = verified_mode.expiration_datetime
+
             relevant_verification = SoftwareSecurePhotoVerification.verification_for_datetime(deadline, verifications)
 
+            # Picking the max verification datetime on each iteration
+            if relevant_verification is not None:
+                recent_verification_datetime = max(
+                    recent_verification_datetime if recent_verification_datetime is not None
+                    else relevant_verification.expiration_datetime, relevant_verification.expiration_datetime
```

Comment:
@wedaly now dates are appearing for approved attempts but other issue is still there which i mentioned that edx course is showing as verified and other course has no message.
And if changes the status of 2nd attempt to denied then black banner appeared. 
please check the image. 
![screen shot 2015-04-10 at 12 59 38 pm](https://cloud.githubusercontent.com/assets/445320/7084005/cd3c1972-df81-11e4-8063-1a990872920b.png)